### PR TITLE
docs: join table type docs

### PR DESCRIPTION
**Pull request by IrakliJani**
docs for: #6414 

### Description:

- [ ] a `joins configuration` section like in metrics here: https://docs.lightdash.com/references/metrics/#metric-configuration
- [ ] This includes an example join with all of the configurations applied
- [ ] It also includes a table with all the configurations described in it
- [ ] These configurations in the table are linked to the sections already in the doc (which explain the configuration options)
- [ ] add a new configuration row + section in the docs for "Specify your join type using `type`"
- [ ] Remove the info at the top of the `joins` doc which says "all joins are LEFT OUTER joins"
- [ ] Add an info block in the `specify your join type` section which says "By default, if no `type` is specified, all joins are `LEFT OUTER` joins"

### CODE REVIEW

File: docs/docs/references/joins.md [hunk 1]
```diff
@@ -78,6 +74,92 @@ Note the following important differences when aliasing models in joins:
...
+          sql_on: ${messages.sent_by} = ${users.user_id}
+```
+
+:::info
+
+By default, if no `type` is specified, all joins are `LEFT OUTER` joins.
+
+:::
+
+## Join types
```
Comment (TuringLovesDeathMetal) [inline docs/docs/references/joins.md:99]:
```suggestion
```

Comment (TuringLovesDeathMetal) [inline docs/docs/references/joins.md:99]:
```suggestion
```

File: docs/docs/references/joins.md [hunk 2]
```diff
@@ -78,6 +74,92 @@ Note the following important differences when aliasing models in joins:
...
+    meta:
+      joins:
+        - join: subscriptions
+          sql_on: ${users.user_id} = ${subscriptions.user_id}
+          type: inner
+```
+
+### left
+
+A left join returns all rows from the left table, and the matching rows from the right table. Non-matching rows will have `NULL` for right table's columns. For example, if you have a `users` table and a `subscriptions` table, a left join would return all users, and the subscription information for users who have a subscription.
```
Comment (TuringLovesDeathMetal) [inline docs/docs/references/joins.md:129]:
```suggestion
A left join returns all rows from the left table (i.e. the model where you're adding the `join` to), and the matching rows from the right table (i.e. the model you've specified in `- join:`). Non-matching rows will have `NULL` for right table's columns. For example, if you have a `users` table and a `subscriptions` table, a left join would return all users, and the subscription information for users who have a subscription.
```

**Review by TuringLovesDeathMetal — COMMENTED** [review]
Just a few tiny changes, once you change those, feel free to merge!

Looks great 😍

**Comment by IrakliJani** [comment]
> Just a few tiny changes, once you change those, feel free to merge!
> 
> Looks great 😍

thanks @TuringLovesDeathMetal !

**Pull request merged**